From 9257f7be5075f0c8e9e8255b42c52a4f990a2a97 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Tue, 22 Apr 2025 16:50:16 +0200
Subject: [PATCH] gpu: fix build with gcc-15

* fixes:
src/3rdparty/chromium/gpu/command_buffer/common/skia_utils.h:29:20: error: 'uint64_t' was not declared in this scope
src/3rdparty/chromium/gpu/command_buffer/common/skia_utils.h:29:28: error: template argument 1 is invalid

src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:49:23: error: 'uint32_t' has not been declared
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:51:33: error: 'uint32_t' has not been declared
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:53:24: error: 'uint32_t' has not been declared
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:55:46: error: 'uint32_t' has not been declared
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:57:27: error: 'uint32_t' has not been declared
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:59:23: error: 'uint32_t' has not been declared
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:66:22: error: 'uint32_t' was not declared in this scope
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:66:61: error: template argument 1 is invalid
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.h:67:3: error: 'GpuFenceEntryMap' does not name a type; did you mean 'GpuFenceEntry'?
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:24:10: error: 'gpu_fence_entries_' was not declared in this scope
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:27:6: error: no declaration matches 'bool gpu::gles2::GpuFenceManager::CreateGpuFence(uint32_t)'
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:43:6: error: no declaration matches 'bool gpu::gles2::GpuFenceManager::CreateGpuFenceFromHandle(uint32_t, gfx::GpuFenceHandle)'
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:66:6: error: no declaration matches 'bool gpu::gles2::GpuFenceManager::IsValidGpuFence(uint32_t)'
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:71:32: error: no declaration matches 'std::unique_ptr<gfx::GpuFence> gpu::gles2::GpuFenceManager::GetGpuFence(uint32_t)'
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:82:6: error: no declaration matches 'bool gpu::gles2::GpuFenceManager::GpuFenceServerWait(uint32_t)'
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:94:6: error: no declaration matches 'bool gpu::gles2::GpuFenceManager::RemoveGpuFence(uint32_t)'
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:102:21: error: 'gpu_fence_entries_' was not declared in this scope
src/3rdparty/chromium/gpu/command_buffer/service/gpu_fence_manager.cc:106:3: error: 'gpu_fence_entries_' was not declared in this scope

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
 chromium/gpu/command_buffer/common/skia_utils.h         | 1 +
 chromium/gpu/command_buffer/service/gpu_fence_manager.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/chromium/gpu/command_buffer/common/skia_utils.h b/chromium/gpu/command_buffer/common/skia_utils.h
index 6508d79a1d4..a9f9528f133 100644
--- a/chromium/gpu/command_buffer/common/skia_utils.h
+++ b/chromium/gpu/command_buffer/common/skia_utils.h
@@ -5,6 +5,7 @@
 #ifndef GPU_COMMAND_BUFFER_COMMON_SKIA_UTILS_H_
 #define GPU_COMMAND_BUFFER_COMMON_SKIA_UTILS_H_
 
+#include <cstdint>
 #include <memory>
 
 #include "base/optional.h"
diff --git a/chromium/gpu/command_buffer/service/gpu_fence_manager.h b/chromium/gpu/command_buffer/service/gpu_fence_manager.h
index ee50d6fddd1..576f6e3bc8a 100644
--- a/chromium/gpu/command_buffer/service/gpu_fence_manager.h
+++ b/chromium/gpu/command_buffer/service/gpu_fence_manager.h
@@ -5,6 +5,7 @@
 #ifndef GPU_COMMAND_BUFFER_SERVICE_GPU_FENCE_MANAGER_H_
 #define GPU_COMMAND_BUFFER_SERVICE_GPU_FENCE_MANAGER_H_
 
+#include <cstdint>
 #include <memory>
 #include <vector>
 
